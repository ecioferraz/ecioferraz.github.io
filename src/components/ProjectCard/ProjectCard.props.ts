type ProjectCardProps = {
  imgSrc: string;
  mainURL: string;
  secondaryURL: string;
  title: string;
};

export default ProjectCardProps;
