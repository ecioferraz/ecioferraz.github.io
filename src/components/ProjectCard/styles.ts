import styled, { css } from 'styled-components';

export const Container = styled.section`
  ${({ theme }) => css`
    background-color: ${theme.colors.secondaryColor};
    border: 1px solid rgba(0, 0, 0, 0.125);
    border-radius: 0.8rem;
    display: flex;
    flex-direction: column;
    margin: 0.8rem;
    max-height: 20rem;
    max-width: 27rem;
    overflow: hidden;

    :hover {
      scale: 1.05;
    }

    @media only screen and ${theme.media.width.lteHuge} {
      max-height: 18rem;
      max-width: 22rem;
    }
  `}
`;

export const ContainerBody = styled.section`
  ${({ theme }) => css`
    align-items: center;
    color: ${theme.colors.primaryColor};
    display: flex;
    flex: 1 1 auto;
    justify-content: space-between;
    padding: 0 1.2rem;
    text-align: center;

    h4 {
      font-size: 1.5rem;
      font-weight: 500;
      margin: 0;
      white-space: nowrap;
    }
  `}
`;
