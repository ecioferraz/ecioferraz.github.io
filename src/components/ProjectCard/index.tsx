import { Code } from '@styled-icons/entypo';
import CustomLink from '../CustomLink';
import ImageCard from '../ImageCard';
import TextCard from '../TextCard';
import ProjectCardProps from './ProjectCard.props';
import * as Styled from './styles';

export default function ProjectCard({
  imgSrc,
  mainURL,
  secondaryURL,
  title,
}: ProjectCardProps) {
  return (
    <Styled.Container title={title}>
      <CustomLink src={mainURL}>
        <ImageCard alt={`Imagem do projeto ${title}`} src={imgSrc} />
      </CustomLink>
      <Styled.ContainerBody>
        <TextCard as="h4" size="small">
          {title}
        </TextCard>
        <CustomLink src={secondaryURL}>
          <Code title='GitHub' />
        </CustomLink>
      </Styled.ContainerBody>
    </Styled.Container>
  );
}
