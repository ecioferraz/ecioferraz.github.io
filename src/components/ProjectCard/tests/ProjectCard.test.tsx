import renderTheme from '../../../styles/renderTheme';
import ProjectCard from '..';

describe('<ProjectCard />', () => {
  it('should render <ProjectCard />', () => {
    const { container } = renderTheme(
      <ProjectCard
        imgSrc="testImgSrc"
        mainURL="testMainURL"
        secondaryURL="testSecondaryURL"
        title="testTitle"
      />,
    );

    expect(container).toMatchSnapshot();
  });
});
