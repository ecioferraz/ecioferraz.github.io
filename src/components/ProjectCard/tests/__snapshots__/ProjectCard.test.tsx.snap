// Vitest Snapshot v1

exports[`<ProjectCard /> > should render <ProjectCard /> 1`] = `
.c5 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 0;
}

.c1 svg {
  color: #000;
  width: 2rem;
}

.c1 svg:hover {
  scale: 1.3;
}

.c2 {
  width: 100%;
}

.c4 {
  font-size: 1.6rem;
}

.c0 {
  background-color: #FFF;
  border: 1px solid rgba(0,0,0,0.125);
  border-radius: 0.8rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin: 0.8rem;
  max-height: 20rem;
  max-width: 27rem;
  overflow: hidden;
}

.c0:hover {
  scale: 1.05;
}

.c3 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #000;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 0 1.2rem;
  text-align: center;
}

.c3 h4 {
  font-size: 1.5rem;
  font-weight: 500;
  margin: 0;
  white-space: nowrap;
}

@media only screen and (max-width:1280px) {
  .c0 {
    max-height: 18rem;
    max-width: 22rem;
  }
}

<div>
  <section
    class="c0"
    title="testTitle"
  >
    <a
      class="c1"
      href="testMainURL"
      target="_blank"
    >
      <img
        alt="Imagem do projeto testTitle"
        class="c2"
        src="testImgSrc"
      />
    </a>
    <section
      class="c3"
    >
      <h4
        class="c4"
      >
        testTitle
      </h4>
      <a
        class="c1"
        href="testSecondaryURL"
        target="_blank"
      >
        <svg
          class="c5"
          fill="currentColor"
          focusable="false"
          role="img"
          viewBox="0 0 20 20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title>
            GitHub
          </title>
          <path
            d="M5.719 14.75a.997.997 0 0 1-.664-.252L-.005 10l5.341-4.748a1 1 0 0 1 1.328 1.495L3.005 10l3.378 3.002a1 1 0 0 1-.664 1.748zm8.945-.002L20.005 10l-5.06-4.498a.999.999 0 1 0-1.328 1.495L16.995 10l-3.659 3.252a1 1 0 0 0 1.328 1.496zm-4.678 1.417 2-12a1 1 0 1 0-1.972-.329l-2 12a1 1 0 1 0 1.972.329z"
          />
        </svg>
      </a>
    </section>
  </section>
</div>
`;
